bundles/org.openhab.binding.liquidcheck/src/main/java/org/openhab/binding/liquidcheck/internal/LiquidCheckHandler.java

```diff
@@ -0,0 +1,199 @@
+        properties.put(PROPERTY_SECURITY_CODE, response.payload.device.security.code);
+        properties.put(PROPERTY_IP, response.payload.wifi.station.ip);
+        properties.put(Thing.PROPERTY_MAC_ADDRESS, response.payload.wifi.station.mac);
+        properties.put(PROPERTY_SSID, response.payload.wifi.accessPoint.ssid);
+        return properties;
+    }
+
+    @Override
+    public void dispose() {
+        ScheduledFuture<?> polling = this.polling;
+        if (null != polling) {
+            polling.cancel(true);
+        }
+    }
+
+    /**
+     * 
+     */
+    private class PollingForData implements Runnable {
+
+        private final LiquidCheckHttpClient client;
+
+        public PollingForData(LiquidCheckHttpClient client) {
+            this.client = client;
+        }
+
+        @Override
+        public void run() {
+            try {
+                String jsonString = client.pollData();
+                CommData response = new Gson().fromJson(jsonString, CommData.class);
+                if (null != response) {
+                    Map<String, String> properties = createPropertyMap(response);
+                    if (!oldProps.equals(properties)) {
+                        oldProps = properties;
+                        updateProperties(properties);
+                    }
+                    updateState(CONTENT_CHANNEL, new QuantityType<>(response.payload.measure.content, Units.LITRE));
+                    updateState(LEVEL_CHANNEL, new QuantityType<>(response.payload.measure.level, SIUnits.METRE));
+                    updateState(RAW_CONTENT_CHANNEL,
+                            new QuantityType<>(response.payload.measure.raw.content, Units.LITRE));
+
+                    updateState(RAW_LEVEL_CHANNEL,
+                            new QuantityType<>(response.payload.measure.raw.level, SIUnits.METRE));
+                    updateState(PUMP_TOTAL_RUNS_CHANNEL, new DecimalType(response.payload.system.pump.totalRuns));
+                    updateState(PUMP_TOTAL_RUNTIME_CHANNEL,
+                            new QuantityType<>(response.payload.system.pump.totalRuntime, Units.SECOND));
+                    if (config.maxContent > 1) {
+                        double fillIndicator = (double) response.payload.measure.content / config.maxContent * 100;
+                        updateState(FILL_INDICATOR_CHANNEL, new QuantityType<>(fillIndicator, Units.PERCENT));
+                    }
+                    if (thing.getStatus().equals(ThingStatus.OFFLINE)) {
+                        updateStatus(ThingStatus.ONLINE);
+                    }
+                } else {
+                    logger.debug("Json is null");
+                }
+            } catch (InterruptedException | TimeoutException | ExecutionException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                logger.error("This went wrong: {}", e.toString());
```

Comment:
Please do not use ERROR level, but WARN level.
Use `e.getMessage()` rather than `e.toString()`

In fact, you should replace these two lines by just this one:
```
updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
```